src/lib/logger.ts

```diff
@@ -0,0 +1,50 @@
+import type { ILevel, ILogger } from '../@types/logger';
+
+/**
+ * A object that handles logging to a given logger based on the logging level
+ */
+export class ManagedLogger {
+  /**
+   * Creates a ManagedLogger that will log only at given severity levels
+   * @param logger a logger where logs will be logged to. If set
+   * to false then logging is disabled
+   * @param level The level(s) of severity to log
+   */
+  constructor(
+    private readonly logger: ILogger | false,
```

Comment:
Given that `ManageLogger` is constructed at the same time as `PrismaSessionStore` there currently is no way to change the logger after construction without constructing a new `PrismaSessionStore`.

The use case here was mostly to have a quick and effective way to disabling the logger for testing so it doesn't spam the console with logs that my testing should be testing against anyways.

You could set `loggerLevel` to an empty array too which would do the same thing however I felt it more expressive to set logger to false to indicate that it is completely disabled despite any logging level. Of course the `loggerLevel` does nothing when the `logger` is set to false but I would think this would be expected by a user as well the `logger` is literally `false` so how could it possible log.

I am open to ideas on how to make these options more ergonomic though.

Reply:
>loggerLevel does nothing when the logger is set to false but I would think this would be expected by a user as well the logger is literally false so how could it possible log.

Fair enough - especially if the primary use case is testing (rather than something end-user facing).

I haven't thought of anything simpler (that doesn't also over-complicate things! ;-)